server/appcap.go

```diff
@@ -91,13 +91,13 @@ func (s *IDPServer) addGrantAccessContext(handler http.HandlerFunc) http.Handler
 
 		who, err = s.lc.WhoIs(r.Context(), remoteAddr)
 		if err != nil {
-			http.Error(w, fmt.Sprintf("Error getting WhoIs: %v", err), http.StatusInternalServerError)
+			writeHTTPError(w, r, http.StatusInternalServerError, "server_error", fmt.Sprintf("Error getting WhoIs: %v", err))
```

Comment:
we might not want to leak the exact error message to the caller here.
I recommend we separate the user-facing error from the internal logged one, like we do in https://github.com/tailscale/tailscale/blob/21c1f4cbff7a76d868faae3edf9a14ad2e5220d4/tsweb/tsweb.go#L886

Reply:
fixed.